base/int.jl

```diff
@@ -43,42 +43,55 @@ copysign(x::Signed, y::Float32) = copysign(x, reinterpret(Int32,y))
 copysign(x::Signed, y::Float64) = copysign(x, reinterpret(Int64,y))
 copysign(x::Signed, y::Real)    = copysign(x, -oftype(x,signbit(y)))
 
-abs(x::Unsigned) = x
-
 """
     abs(x)
 
-The absolute value of `x`.  When `abs` is applied to signed integers,
-overflow may occur, resulting in the return of a negative value.  This
-overflow occurs only when `abs` is applied to the minimum
-representable value of a signed integer.  That is when `x ==
-typemin(typeof(x))`, `abs(x) == x`, not `-x` as might be expected.
+The absolute value of `x`.
+
+When `abs` is applied to signed integers, overflow may occur,
+resulting in the return of a negative value. This overflow occurs only
+when `abs` is applied to the minimum representable value of a signed
+integer. That is, when `x == typemin(typeof(x))`, `abs(x) == x < 0`,
+not `-x` as might be expected.
 """
+function abs end
+
+abs(x::Unsigned) = x
 abs(x::Signed) = flipsign(x,x)
 
 """
     Base.checked_abs(x)
 
-The absolute value of `x`, with overflow error trapping where applicable.
+The absolute value of `x`, with overflow error trapping where
+applicable. Otherwise, behaves as `-x`, though the overflow protection
+may impose a perceptible performance penalty.
+
 For types for which no overflow can happen during `abs`, this is equivalent
 to `abs(x)`.
 """
 function checked_abs end
 
+checked_abs(x::Unsigned) = abs(x)
+function checked_abs{T<:Union{Int8,Int16,Int32,Int64,Int128}}(x::T)
+    x == typemin(T) && throw(OverflowError())
+    abs(x)
+end
+
 """
-    Base.checked_abs(x::Signed)
+    Base.checked_neg(x)
```

Comment:
add signature to rst then rerun genstdlib.jl if you want the docstring to show up there